doc/src/site/sphinx/ml/sw_xgboost.rst

```diff
@@ -118,18 +118,23 @@ XGBoost Memory Configuration
 H2O XGBoost uses additionally to Java memory, off-heap memory. This means that it requires some additional memory
 available on the system.
 
-When running on YARN, please make sure to set the ``memoryOverhead`` so XGBoost has enough memory. On Spark, the following
-properties might be set
+When running on YARN or Kubernetes, please make sure to set the ``SUBST_EXECUTOR_MEMORY_OVERHEAD`` so XGBoost has enough
+native memory on executors. It's recommended to set the property to 12O% of the value set in ``spark.executor.memory``.
 
-- ``spark.yarn.am.memoryOverhead`` - in case of YARN Cluster deployment
-- ``spark.yarn.driver.memoryOverhead`` - in case of YARN client and other deployments
-- ``spark.yarn.executor.memoryOverhead`` - in all deployment scenarios
+Note: ``SUBST_EXECUTOR_MEMORY_OVERHEAD`` must be set in MiB.
 
-On YARN, the container size is determined by ``application_memory * memory_overhead``. Therefore, by specifying the
-overhead, we are also allocating some additional off-heap memory which XGBoost can use.
+Example
+```````
+If you set ``spark.executor.memory`` to ``10g``, ``SUBST_EXECUTOR_MEMORY_OVERHEAD`` should be set to ``12288``.
+The size of the corresponding YARN or Kubernetes container will be at least 22 GiB.
 
-In Spark Standalone Mode or IBM Conductor environment, please make sure to configure the following configurations:
+Note: In case of Pysparkling, the YARN container will be bigger about the memory required by the Python process.
 
 
-- ``spark.memory.offHeap.enabled=true``
-- ``spark.memory.offHeap.size=4G`` - example of setting this property to 4G of off-heap memory
\ No newline at end of file
+Memory Overhead on Spark driver
+```````````````````````````````
+If you set the SW property ``spark.ext.h2o.rest.api.based.client`` to ``false``, you should also set the following properties
```

Comment:
Does this refer to the "H2O client"?

Reply:
yes, you're right. By default, H2O client is disabled since the version `3.30.0.1-1`.

Reply:
ok, I think I get it, please confirm that my understanding is correct:

Setting `spark.ext.h2o.rest.api.based.client` to `false` enables the "H2O client mode" and you thus need set overhead also for the process where the client code is running, right?

Reply:
Just clarification -> `spark.ext.h2o.rest.api.based.client` is valid only when running Sparkling Water in Scala/Java. There is no option to enable H2O client in R/Python, no matter what this option is set to. Python/R is always using Rest client

Reply:
@jakubhava What about SW Python API it's backed with Scala objects. the same holds for `H2OContext`.

Reply:
Yup, it is backed by Scala objects, but no H2O client running. There is no H2O client on Py/R at all

Reply:
ok, i see the property `spark.ext.h2o.rest.api.based.client` manually hardcoded to "true"

Reply:
Now, I'm wondering if the memory overhead is needed on the driver at all. If I'm not mistaken all the Rest API requests goes to the H2O client if it's enabled. Otherwise, it goes to the leader node. 

@michalkurka As far I remember, XGboost model building was performed on the H2O client only when a specific Java method for triggering the model training was called. I remember at some point that this could be omitted by calling a different one. Now all commands go through Rest API (`/3/ModelBuilders/`). Do you think that H2O client still will be part of model training?

Reply:
The client will definitely need some native memory if invoked from Java/Scala. If REST API is used, it will be forwarded to an H2O node instead of the client.

The client is never part of the training, but does access the native memory to retrieve the booster.

Reply:
I see, thank you both! I will add a note that this is relevant only the scala API.